### DISCUSSION TITLE
Github Changelog Compare URL

### DISCUSSION THREAD

**Discussion by thumDer** [Nice3point/RevitTemplates#58] [General]
I've found that there is a method in the github ci target that should create a compare for the release automatically: 
https://github.com/Nice3point/RevitTemplates/blob/85a2b5dff6cd5ef13c4ae3f0d8a43437307fcdca/build/Build.CI.GitHub.cs#L70
For me this workflow is triggered on master branch pushes, and it never creates the changelog, the method returns early, because `if (latestTag == GitRepository.Commit) return;`
How this supposed to work? Am I doing something wrong?
Thanks!

**Comment by Nice3point** [comment]
`if (latestTag == GitRepository.Commit) return;` checks if there is an existing release. if `latestTag == GitRepository.Commit` there were no releases and there is no reason to make a link to compare versions. if there were releases `latestTag` will contain the version of the latest release.

**Reply by thumDer** [reply]
Yes, that is what I thought, however for me `latastTag` contains the release that is getting created, and not the previous.

**Reply by thumDer** [reply]
@Nice3point could you please check if it works for you as you would expect?

**Reply by Nice3point** [reply]
@thumDer add `fetch-depth: 0` to the Github workflow. 

Sample:
https://github.com/jeremytammik/RevitLookup/blob/dev/.github/workflows/Nuke.yml
https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches

I will add this property in the new patch, by default

**Reply by thumDer** [reply]
great, thanks, will give it a try!

**Comment by Nice3point** [comment]
Drafted for 2026
